### PR TITLE
Added tests for various view models used by the drop in UI.

### PR DESCRIPTION
**Pull request by cafesilencio**
### Description
Added tests for various view models used by the drop in UI.

### CODE REVIEW

File: libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/component/maneuver/ManeuverProcessorTest.kt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        val result = ManeuverProcessor.ProcessVisibility(
+            NavigationState.Empty,
+            ExpectedFactory.createError(mockk<ManeuverError>())
+        ).process()
+
+        assertFalse(result.isVisible)
+    }
+
+    @Test
+    fun processVisibility_navigationState_empty_maneuverIsValue() {
```
Comment (kmadsen) [inline libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/component/maneuver/ManeuverProcessorTest.kt:29]: nit: junit5 looks like it is trying to parameterize tests more. but sealed classes may have issues. was looking around and found this https://github.com/dpreussler/junit5-kotlin but not sure we really want to integrate that

short term maybe we could just do it ourselves. my tests aren't working though

 ``` kotlin
    @Test
    fun errors_are_not_visible() {
        val navigationStates = NavigationState::class.sealedSubclasses
            .map { it.objectInstance as NavigationState }
        navigationStates.onEach { navigationState ->
            val result = ManeuverProcessor.ProcessVisibility(
                navigationState,
                ExpectedFactory.createError(mockk())
            ).process()

            assertFalse("$navigationState", result.isVisible)
        }
    }
```

**Review by kmadsen — APPROVED** [review]

**Comment by kmadsen** [comment]
🤔 

```
java.lang.NoClassDefFoundError: jdk/internal/reflect/GeneratedSerializationConstructorAccessor1
	at jdk.internal.reflect.GeneratedSerializationConstructorAccessor1.newInstance(Unknown Source)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.io.ObjectStreamClass.newInstance(ObjectStreamClass.java:1092)
	at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2180)
	at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1679)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:493)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:451)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.deserializeWorker(SystemApplicationClassLoaderWorker.java:153)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:121)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
Caused by: java.lang.ClassNotFoundException: jdk.internal.reflect.GeneratedSerializationConstructorAccessor1
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:471)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 12 more
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':libnavui-dropin:testDebugUnitTest'.
> No tests found for given includes: [com.mapbox.navigation.dropin.component.maneuver.ManeuverProcessorTest](filter.includeTestsMatching)
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
* Get more help at https://help.gradle.org
* ```

**Comment by cafesilencio** [comment]
> java.lang.NoClassDefFoundError: jdk/internal/reflect/GeneratedSerializationConstructorAccessor1

This didn't happen at the time I created the PR. I think some of the recent updates in gradle etc. have caused some complications.  I'll have to make some adjustments to get these tests working again.

File: libnavui-dropin/build.gradle [hunk 2]
```diff
@@ -33,6 +33,11 @@ android {
     testOptions {
         unitTests.returnDefaultValues = true
         unitTests.includeAndroidResources = true
+        unitTests.all {
+            jacoco {
+                excludes = ['jdk.internal.*']
+            }
+        }
```
Comment (Zayankovsky) [inline libnavui-dropin/build.gradle:40]: Rebasing on top of `main` probably should have been enough, because we already have that setup in jacoco.gradle.

Comment (cafesilencio) [reply libnavui-dropin/build.gradle]: yep, rebased and removed it from the gradle file.

**Review by abhishek1508 — APPROVED** [review]

**Pull request merged**